gateway/src/shard/inflater.rs

```diff
@@ -0,0 +1,109 @@
+use std::convert::TryInto;
+
+use flate2::{Decompress, DecompressError, FlushDecompress};
+use log::trace;
+
+const ZLIB_SUFFIX: [u8; 4] = [0x00, 0x00, 0xff, 0xff];
+const INTERNAL_BUFFER_SIZE: usize = 32 * 1024;
+const COUNTDOWN: u64 = 255;
```

Comment:
Adding to the comment below, this can just be a u8. Let's remove this and move the definition to `Inflater::new`, using the `u8::max_value()` constant